src/jazayeri_lab_to_nwb/watters/wattersrecordinginterface.py

```diff
@@ -89,36 +28,55 @@ def __init__(
         probe_metadata_file: Optional[FilePathType] = None,
         probe_name: str = "vprobe",
         probe_key: Optional[str] = None,
     ):
-        traces = np.memmap(file_path, dtype=dtype, mode="r").reshape(-1, channel_count)
         source_data = {
-            "traces_list": [traces],
+            "file_paths": [file_path],
             "sampling_frequency": sampling_frequency,
+            "num_channels": channel_count,
             "t_starts": [t_start],
             "channel_ids": channel_ids,
+            "gain_to_uV": gain_to_uv,
+            "offset_to_uV": offset_to_uv,
+            "dtype": dtype,
         }
         super().__init__(verbose=verbose, es_key=es_key, **source_data)
-        if gain_to_uv is not None:
-            if len(gain_to_uv) == 1:
-                gain_to_uv = np.full((channel_count,), gain_to_uv[0], dtype=float)
-            else:
-                assert len(gain_to_uv) == channel_count, (
-                    f"There are {channel_count} channels " f"but `gain_to_uv` has length {len(gain_to_uv)}"
-                )
-                gain_to_uv = np.array(gain_to_uv, dtype=float)
-            self.recording_extractor.set_property("gain_to_uV", gain_to_uv)
-        self.probe_metadata_file = probe_metadata_file
+
+        # this is used for metadata naming
         self.probe_name = probe_name
-        self.probe_key = probe_key
 
-        self.electrode_metadata = None
-        if self.probe_metadata_file is not None and self.probe_key is not None:
-            self.electrode_metadata = add_electrode_locations(
-                self.recording_extractor, self.probe_metadata_file, self.probe_name, self.probe_key
-            )
+        # add probe information
+        probe_metadata = None
+        if probe_metadata_file is not None and probe_key is not None:
+            with open(probe_metadata_file, "r") as f:
+                all_probe_metadata = json.load(f)
+            for entry in all_probe_metadata:
+                if entry["label"] == probe_key:
+                    probe_metadata = entry
+
+        if probe_metadata is not None and "electrodes_locations" in probe_metadata:
+            # Grab electrode position from metadata
+            locations_array = np.array(probe_metadata["electrodes_locations"])
+            ndim = locations_array.shape[1]
+            probe = pi.Probe(ndim=ndim)
+            probe.set_contacts(locations_array)
+        else:
+            # Generate V-probe geometry: 64 channels arranged vertically with 50 um spacing
+            probe = pi.generate_linear_probe(num_elec=64, ypitch=50)
+        probe.name = probe_name
+
+        # set probe to interface recording
+        self.set_probe(probe)
+
+        # set group_name property to match electrode group name in metadata
+        self.recording_extractor.set_property(
+            key="group_name",
+            values=[probe_name] * len(self.recording_extractor.channel_ids),
+        )
```

Comment:
Doesn't Jeremy's function do this bit already? https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/datainterfaces/ecephys/baserecordingextractorinterface.py#L249